tests/unit/cluster/failure-marking.tcl

```diff
@@ -0,0 +1,87 @@
+proc check_node_marked_fail {ref_node_index instance_id_to_check} {
+    set nodes [get_cluster_nodes $ref_node_index]
+
+    foreach n $nodes {
+        if {[dict get $n id] eq $instance_id_to_check} {
+            return [cluster_has_flag $n fail]
+        }
+    }
+    fail "Unable to find instance id in cluster nodes. ID: $instance_id_to_check"
+}
+
+proc check_node_marked_pfail {ref_node_index instance_id_to_check} {
+    set nodes [get_cluster_nodes $ref_node_index]
+
+    foreach n $nodes {
+        if {[dict get $n id] eq $instance_id_to_check} {
+            return [cluster_has_flag $n fail?]
+        }
+    }
+    fail "Unable to find instance id in cluster nodes. ID: $instance_id_to_check"
+}
```

Comment:
maybe you could parameterize the value to compare with `fail` / `fail?` ? code seems duplicate.

Reply:
Made wait_for_x functions for pfail and fail, then parameterized the check; I feel it is important to have both a "wait_node_marked_pfail" as well as a "wait_node_marked_fail" as it is not super obvious that the flag for pfail is "fail?"